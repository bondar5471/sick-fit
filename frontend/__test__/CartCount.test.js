import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import CartCount from '../components/CardCount'

describe('<CartCount/>', () => {
  const wrapper = shallow(<CartCount count={10}/>)
  it('renders', () => {
    shallow(<CartCount count={10} />)
  })

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  it('update via props ', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.setProps({count: 15})
    expect(toJson(wrapper)).toMatchSnapshot()
  });
});