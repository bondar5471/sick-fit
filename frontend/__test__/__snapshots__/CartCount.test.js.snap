// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> matches the snapshot 1`] = `
<CardCount__AnimationStyle>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CardCount__Dot>
        10
      </CardCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CardCount__AnimationStyle>
`;

exports[`<CartCount/> update via props  1`] = `
<CardCount__AnimationStyle>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CardCount__Dot>
        10
      </CardCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CardCount__AnimationStyle>
`;

exports[`<CartCount/> update via props  2`] = `
<CardCount__AnimationStyle>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="15"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CardCount__Dot>
        15
      </CardCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CardCount__AnimationStyle>
`;
